packages/flutter/lib/src/animation/curves.dart

```diff
@@ -267,6 +273,526 @@ class Cubic extends Curve {
+      }
+      count--;
+    }
+    return mid;
+  }
+}
+
+/// A curve that passes smoothly through the given control points using a
+/// centripetal Catmull-Rom spline.
+///
+/// When this curve is evaluated with [transform], the values will interpolate
+/// smoothly from one control point to the next, passing through (0.0, 0.0), the
+/// given points, and then (1.0, 1.0).
+///
+/// {@macro flutter.animation.curves.catmull_rom_description}
+///
+/// This class uses a centripetal Catmull-Rom curve (a [CatmullRomSpline]) as
+/// its internal representation. The term centripetal implies that it won't form
+/// loops or self-intersections within a single segment, and corresponds to a
+/// Catmull-Rom α (alpha) value of 0.5.
+///
+/// See also:
+///
+///  * [CatmullRomSpline], the 2D spline that this curve uses to generate its values.
+///  * A Wikipedia article on [centripetal Catmull-Rom splines](https://en.wikipedia.org/wiki/Centripetal_Catmull%E2%80%93Rom_spline).
+///  * [new CatmullRomCurve] for a description of the constraints put on the
+///    input control points.
+///  * This [paper on using Catmull-Rom splines](http://faculty.cs.tamu.edu/schaefer/research/cr_cad.pdf).
+class CatmullRomCurve extends Curve {
+  /// Constructs a centripetal [CatmullRomCurve].
+  ///
+  /// It takes a list of two or more points that describe the points that the
+  /// curve must pass through. See [controlPoints] for a description of the
+  /// restrictions placed on control points.
+  ///
+  /// The optional [tension] argument controls how tightly the curve approaches
+  /// the given `points`. It must be in the range 0.0 to 1.0, inclusive. It
+  /// defaults to 0.0, which provides the smoothest curve. A value of 1.0
+  /// is equivalent to a linear interpolation between points.
+  ///
+  /// All of the arguments must not be null.
+  ///
+  /// See also:
+  ///
+  ///  * This [paper on using Catmull-Rom splines](http://faculty.cs.tamu.edu/schaefer/research/cr_cad.pdf).
+  CatmullRomCurve(this.controlPoints, {this.tension = 0.0})
+      : assert(tension != null),
+        assert(() {
+          return validateControlPoints(
+            controlPoints,
+            tension: tension,
+            reasons: _assertReasons..clear(),
+          );
+        }(), 'control points $controlPoints could not be validated:\n  ${_assertReasons.join('\n  ')}'),
+        // Pre-compute samples so that we don't have to evaluate the spline's inverse
+        // all the time in transformInternal.
+        _precomputedSamples = CatmullRomSpline(
+          // Force the first and last control points for the spline to be (0, 0)
+          // and (1, 1), respectively.
+          <Offset>[Offset.zero, ...controlPoints, const Offset(1.0, 1.0)],
+          tension: tension,
+        ).generateSamples(start: 0.0, end: 1.0, tolerance: 1e-12);
+
+  /// A static accumulator for assertion failures. Not used in release mode.
+  static final List<String> _assertReasons = <String>[];
```

Comment:
prefix this with "debug" to indicate that it will not be used in release mode?

Reply:
This static seems a little hacky.

Reply:
It is, but I don't know of another way to get it to appear in the assert details.  I used to have it just print out when it was going to fail, but then it printed the details before the assert fired, which was not desirable either.